### DISCUSSION TITLE
HelmRelease upgrade of snapshot version?

### DISCUSSION THREAD

**Discussion by mprusinski** [fluxcd/flux2#2776] [Q&A]
Hi,

I would like to use flux to deploy of a helm chart to let's say "dev" environment. All of the artifacts (docker images, helm packages) used there are in their SNAPSHOT versions (the version of the artifact is not bumped, but the artifacts themselves are being overwritten in the artifact repository after each build under one and the same version all over again).

For example lets say I have a service myService with 2 artifacts:
- docker image called myServiceDockerImage:1.0.0-SNAPSHOT
- helm chart called myServiceChart:1.0.0-SNAPSHOT

I've created flux HelmRelease object with this version fixed and the flux is able to deploy the chart and the service below to my k8s cluster.

Now, the problem occurs when a change is delivered to the myService code, which results in building new artifacts (again: 
 myServiceDockerImage:1.0.0-SNAPSHOT and myServiceChart:1.0.0-SNAPSHOT).

Flux is not detecting those new artifacts (because HelmChart revision didn't change) and in result it is not upgrading the chart.

Does anyone know if I somehow can force flux to do the upgrade in such case?

Thanks for any feedback.

**Comment by kingdonb** [comment]
If your tags are expressed in semver notation like this,

```
semver: >=1.0.0
  or
semver: 1.x.x
```
then Flux will only choose release versions (not PRErelease semver versions) in the range as specified.

The way to make Flux detect prerelease tags is to include a prerelease part in your filter like this:

```
semver: 1.x.x-0
```

For Helm Controller this is equivalent to setting the `--devel` flag for `helm install`

Hope this helps!
